src/input/code_templates/c/static_code/exi_basetypes.c.jinja

```diff
@@ -97,4 +117,80 @@ int exi_basetypes_convert_64_from_unsigned(exi_unsigned_t* exi_unsigned, uint64_
 
     return EXI_ERROR__NO_ERROR;
 }
+
+int exi_basetypes_convert_from_signed(const exi_signed_t* exi_signed, int32_t* value, size_t max_octets)
+{
+    uint32_t uValue = 0;
+    int res = exi_basetypes_convert_from_unsigned(&exi_signed->data, &uValue, max_octets);
+    *value = (exi_signed->is_negative == 0) ? uValue : -uValue;
+    return res;
+}
+
+int exi_basetypes_convert_64_from_signed(const exi_signed_t* exi_signed, int64_t* value)
+{
+    uint64_t uValue = 0;
+    int res = exi_basetypes_convert_64_from_unsigned(&exi_signed->data, &uValue);
+    *value = (exi_signed->is_negative == 0) ? uValue : -uValue;
+    return res;
+}
+
+int exi_basetypes_convert_bytes_from_unsigned(const exi_unsigned_t* exi_unsigned, uint8_t* data, size_t* data_len, size_t data_size)
```

Comment:
My rule of thumb is if it's larger than a register you probably don't want to copy it and currently it's gotten quite big relatively speaking. Though that depends on the device of course, a CPU with AX256 will fit it just fine, but armv5 won't.

The function will take 4-8 bytes instead of this 25+byte object.

This will likely be more efficient on this lowest common denominator, though I admit I didn't test it.

Reply:
I agree with your reasoning.